internal/appsec/listener/grpcsec/grpc.go

```diff
@@ -100,35 +103,36 @@ func (l *wafEventListener) onEvent(op *types.HandlerOperation, handlerArgs types
 		return
 	}
 
-	// OnUserIDOperationStart happens when appsec.SetUser() is called. We run the WAF and apply actions to
-	// see if the associated user should be blocked. Since we don't control the execution flow in this case
-	// (SetUser is SDK), we delegate the responsibility of interrupting the handler to the user.
-	dyngo.On(op, func(userIDOp *sharedsec.UserIDOperation, args sharedsec.UserIDOperationArgs) {
-		values := make(map[string]any, 1)
-		for addr := range l.addresses {
-			if addr == UserIDAddr {
-				values[UserIDAddr] = args.UserID
+	// Listen to the UserID address if the WAF rules are using it
+	if l.isSecAddressListened(UserIDAddr) {
+		// UserIDOperation happens when appsec.SetUser() is called. We run the WAF and apply actions to
+		// see if the associated user should be blocked. Since we don't control the execution flow in this case
+		// (SetUser is SDK), we delegate the responsibility of interrupting the handler to the user.
+		dyngo.On(op, func(userIDOp *sharedsec.UserIDOperation, args sharedsec.UserIDOperationArgs) {
+			values := map[string]any{
+				UserIDAddr: args.UserID,
 			}
-		}
-		wafResult := shared.RunWAF(wafCtx, waf.RunAddressData{Persistent: values}, l.config.WAFTimeout)
-		if wafResult.HasActions() || wafResult.HasEvents() {
-			for _, id := range wafResult.Actions {
-				if a, ok := l.actions[id]; ok && a.Blocking() {
-					code, err := a.GRPC()(map[string][]string{})
-					dyngo.EmitData(userIDOp, types.NewMonitoringError(err.Error(), code))
+			wafResult := shared.RunWAF(wafCtx, waf.RunAddressData{Persistent: values}, l.config.WAFTimeout)
+			if wafResult.HasActions() || wafResult.HasEvents() {
+				for _, id := range wafResult.Actions {
+					if a, ok := l.actions[id]; ok && a.Blocking() {
+						code, err := a.GRPC()(map[string][]string{})
+						dyngo.EmitData(userIDOp, types.NewMonitoringError(err.Error(), code))
+					}
 				}
+				shared.AddSecurityEvents(op, l.limiter, wafResult.Events)
+				log.Debug("appsec: WAF detected an authenticated user attack: %s", args.UserID)
 			}
-			shared.AddSecurityEvents(op, l.limiter, wafResult.Events)
-			log.Debug("appsec: WAF detected an authenticated user attack: %s", args.UserID)
-		}
-	})
+		})
+	}
 
-	// The same address is used for gRPC and http when it comes to client ip
-	values := make(map[string]any, 1)
-	for addr := range l.addresses {
-		if addr == HTTPClientIPAddr && handlerArgs.ClientIP.IsValid() {
-			values[HTTPClientIPAddr] = handlerArgs.ClientIP.String()
-		}
+	values := make(map[string]any, 2) // 2 because the method and client ip addresses are commonly present in the rules
+	if l.isSecAddressListened(GRPCServerMethodAddr) {
+		values[GRPCServerMethodAddr] = handlerArgs.Method
+		// Note that this address is passed asap for the passlist, which are created per grpc method
```

Comment:
Completely optional, but it feels odd to me that the comment applies to what's above it... usually it's what's below it.


```suggestion
		// Note that this address is passed asap for the passlist, which are created per grpc method
		values[GRPCServerMethodAddr] = handlerArgs.Method
```